### ISSUE TITLE
Slightly change the design

### ISSUE THREAD

**Issue by ghost** [andreaferretti/interfaced#5]
Just a proposal, would this [change](https://github.com/notTito/nim-100days/blob/master/vtables/vshapes.nim) make any sense? It basically avoids the intermediate anonymous proc but procs need to accept a ptr to an object.

**Comment by krux02** [comment]
can you summarize what you changed?

**Comment by ghost** [comment]
So every object has its own vtable type and  its casted to ShapeVtable type. This way no intermediate anonymous proc is created.
